subsys/bluetooth/mesh/light_ctl_internal.h

```diff
@@ -17,21 +17,35 @@
 static inline uint16_t lvl_to_temp(struct bt_mesh_light_temp_srv *srv,
 				   int16_t lvl)
 {
-	return srv->temp_range.min +
-	       (lvl + ((UINT16_MAX / 2) + 1)) *
-		       (srv->temp_range.max - srv->temp_range.min) / UINT16_MAX;
+	return srv->range.min + (lvl + ((UINT16_MAX / 2) + 1)) *
+					(srv->range.max - srv->range.min) /
+					UINT16_MAX;
 }
 
 static inline int16_t temp_to_lvl(struct bt_mesh_light_temp_srv *srv,
 				  uint16_t raw_temp)
 {
-	uint16_t temp = MAX(raw_temp, srv->temp_range.min);
+	uint16_t temp = MAX(raw_temp, srv->range.min);
```

Comment:
Looks like this string does not have a sense. The result of this will be rewritten by the next line always. Is it ok?

Reply:
I think the point is to ensure that the incoming raw temperature stays within the current range restrictions. `temp` will only be set to `range.min` if `raw_temp` is lower than the permitted range. Likewise, `temp` will only be set to `range.max` if `raw_temp` is higher than the permitted range.

Reply:
`temp = MIN(temp, srv->range.max);` I guess?

Reply:
@alxelax any comment to @Andrewpini? I believe he's right

Reply:
yep, I agree with @Andrewpini. It should be fixed as he suggested.

Reply:
I'm not really sure what the suggestion was here, but it could be converted to a clamp, so I did that.

Reply:
It should be:
`
uint16_t temp = MAX(raw_temp, srv->range.min);
temp = MIN(temp, srv->range.max);
`